core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/workflow/WorkflowVersionResource.scala

```diff
@@ -30,7 +33,98 @@ import scala.collection.convert.ImplicitConversions.`collection AsScalaIterable`
 object WorkflowVersionResource {
   final private lazy val context = SqlServer.createDSLContext()
   final private val workflowDao = new WorkflowDao(context.configuration)
+  // constant to indicate versions should be aggregated if they are within the specified time limit
+  private final val AGGREGATE_TIME_LIMIT_MILLSEC =
+    AmberUtils.amberConfig.getInt("user-sys.version-time-limit") * 60000
+  // list of Json keys in the diff patch that are considered UNimportant
+  private final val VERSION_UNIMPORTANCE_RULES = List("/operatorPositions/")
 
+  /**
+    * This function gives a label to each version whether it is significant or not based on a few rules
+    * The reason why it is computed AFTER retrieving the list of versions is due to multiple reasons:
+    * * 1. minimize the changes to the database.
+    * * 2. since the frontend sends persisting workflow request often, we don't want to slow down the
+    * insertion to DB because of computing the version's importance especially because the request is
+    * async, the versions can quickly become inconsistent if there is delay.
+    * * 3. The rules can be changed in the future so we want this logic to be changed flexibly.
+    * @param versions
+    * @return
+    */
+  private def encodeVersionImportance(
```

Comment:
Agreed. I initially implemented it using `Byte` type but then changed it to `Boolean` because it was an overkill and I noticed I only used 0 and 1 as values of importance.